scripts/run-integration-tests.sh

```diff
@@ -23,55 +42,62 @@ REPORT_DIR=${TEST_DIR}/report
 CLUSTER_ID=${CLUSTER_ID:-$RANDOM}
 CLUSTER_NAME=cni-test-$CLUSTER_ID
 TEST_CLUSTER_DIR=/tmp/cni-test/cluster-$CLUSTER_NAME
+CLUSTER_MANAGE_LOG_PATH=$TEST_CLUSTER_DIR/cluster-manage.log
 CLUSTER_CONFIG=${CLUSTER_CONFIG:-${TEST_CLUSTER_DIR}/${CLUSTER_NAME}.yaml}
 SSH_KEY_PATH=${SSH_KEY_PATH:-${TEST_CLUSTER_DIR}/id_rsa}
 KUBECONFIG_PATH=${KUBECONFIG_PATH:-${TEST_CLUSTER_DIR}/kubeconfig}
 
 # shared binaries
-TESTER_DOWNLOAD_URL=https://github.com/aws/aws-k8s-tester/releases/download/v0.4.3/aws-k8s-tester-v0.4.3-$OS-$ARCH
 TESTER_DIR=${TESTER_DIR:-/tmp/aws-k8s-tester}
 TESTER_PATH=${TESTER_PATH:-$TESTER_DIR/aws-k8s-tester}
-AUTHENTICATOR_PATH=${AUTHENTICATOR_PATH:-/tmp/aws-k8s-tester/aws-iam-authenticator}
-KUBECTL_PATH=${KUBECTL_PATH:-/tmp/aws-k8s-tester/kubectl}
+AUTHENTICATOR_PATH=${AUTHENTICATOR_PATH:-$TESTER_DIR/aws-iam-authenticator}
+KUBECTL_PATH=${KUBECTL_PATH:-$TESTER_DIR/kubectl}
+
+# double-check all our preconditions and requirements have been met
+check_is_installed docker
+check_is_installed aws
+check_aws_credentials
+ensure_aws_k8s_tester
+
+AWS_ACCOUNT_ID=${AWS_ACCOUNT_ID:-$(aws sts get-caller-identity --query Account --output text)}
+AWS_ECR_REGISTRY=${AWS_ECR_REGISTRY:-"$AWS_ACCOUNT_ID.dkr.ecr.$AWS_REGION.amazonaws.com"}
+AWS_ECR_REPO_NAME=${AWS_ECR_REPO_NAME:-"amazon-k8s-cni"}
+IMAGE_NAME=${IMAGE_NAME:-"$AWS_ECR_REGISTRY/$AWS_ECR_REPO_NAME"}
+IMAGE_VERSION=${IMAGE_VERSION:-$(git describe --tags --always --dirty)}
+
+if [[ "$BUILD" = true ]]; then
+    # `aws ec2 get-login` returns a docker login string, which we eval here to
+    # login to the ECR registry
+    eval $(aws ecr get-login --region $AWS_REGION --no-include-email) >/dev/null 2>&1
+    ensure_ecr_repo "$AWS_ACCOUNT_ID" "$AWS_ECR_REPO_NAME"
+    make docker IMAGE=$IMAGE_NAME VERSION=$IMAGE_VERSION
```

Comment:
This doesn't pull source code from given tag(version) right?

Reply:
No, it doesn't. It runs `docker build -t aws-vpc-cni:$IMAGE_VERSION`. If IMAGE_VERSION envvar has been overridden, we should probably set the BUILD envvar to false automatically.

Reply:
Either that, or check that IMAGE_VERSION is overridden and if so, do a `git checkout $IMAGE_VERSION` here to switch the working tree to that git tag.

Reply:
using git-checkout makes sense. Can you help me understand how disabling BUILD will help? Do you mean to say, if CNI image is already available in my repo then disabling BUILD will help?

Reply:
yes, exactly. if you've already pushed, say, a v1.5.5 CNI image built from that git tag/branch, then setting BUILD=false will prevent the run-integration-tests.sh script from `docker push`ing an image and the eks-k8s-tester binary will simply pull the image that was in the ECR repo for v1.5.5

Reply:
Ah ok makes sense. Thanks for clarifying.

Reply:
In short, I need to make some changes to this script to facilitate the not-building-from-local-checkout scenarios...